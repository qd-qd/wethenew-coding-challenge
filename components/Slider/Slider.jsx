import PropTypes from 'prop-types';
import styles from './Slider.module.css';

const Slider = ({ images }) => (
  <section className={styles.slider} data-testid="slider">
    <div className={styles.slides}>
      {images.map(image => (
        <div
          className={styles.slide}
          key={image}
          data-testid={`slide-${image}`}
        >
          <img src={image} alt="" className={styles.image} />
        </div>
      ))}
    </div>
  </section>
);

Slider.propTypes = {
  images: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default Slider;
