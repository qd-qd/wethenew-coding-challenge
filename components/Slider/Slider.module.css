.slider {
  --slider-square-size: 500px;
  width: var(--slider-square-size);
  text-align: center;
  overflow: hidden;
}

.slides {
  display: flex;
  overflow-x: auto;
  scroll-snap-type: x mandatory;
  scroll-behavior: smooth;
  -webkit-overflow-scrolling: touch;
}

.slides::-webkit-scrollbar {
  width: 10px;
  height: 10px;
}
.slides::-webkit-scrollbar-thumb {
  background: black;
  border-radius: 10px;
}
.slides::-webkit-scrollbar-track {
  background: transparent;
}

.slide {
  position: relative;
  scroll-snap-align: start;
  flex-shrink: 0;
  width: var(--slider-square-size);
  height: var(--slider-square-size);
  margin-right: 50px;
  border-radius: 10px;
  transform-origin: center center;
  transform: scale(1);
  transition: transform 0.5s;

  display: flex;
  justify-content: center;
  align-items: center;
}

.image {
  object-fit: cover;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}
