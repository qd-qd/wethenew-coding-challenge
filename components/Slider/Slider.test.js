import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Slider from './Slider';

const images = [
  'https://image.com/foo.jpg',
  'https://image.com/toto.jpg',
  'https://image.com/tata.jpg',
];

describe('Slider', () => {
  test('displays greeting', () => {
    render(<Slider images={images} />);
    expect(screen.getByTestId('slider')).toBeDefined();

    images.forEach(image => {
      expect(screen.getByTestId(`slide-${image}`)).toBeDefined();
      const imgElement = screen.getByTestId(`slide-${image}`).firstChild;
      expect(imgElement).toHaveAttribute('class');
      expect(imgElement).toHaveAttribute('src');
    });
  });

  test("doesn't break snapshot", () => {
    const { container } = render(<Slider images={images} />);
    expect(container).toMatchSnapshot();
  });
});
