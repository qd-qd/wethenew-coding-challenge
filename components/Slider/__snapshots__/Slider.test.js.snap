// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider doesn't break snapshot 1`] = `
<div>
  <section
    class="slider"
    data-testid="slider"
  >
    <div
      class="slides"
    >
      <div
        class="slide"
        data-testid="slide-https://image.com/foo.jpg"
      >
        <img
          alt=""
          class="image"
          src="https://image.com/foo.jpg"
        />
      </div>
      <div
        class="slide"
        data-testid="slide-https://image.com/toto.jpg"
      >
        <img
          alt=""
          class="image"
          src="https://image.com/toto.jpg"
        />
      </div>
      <div
        class="slide"
        data-testid="slide-https://image.com/tata.jpg"
      >
        <img
          alt=""
          class="image"
          src="https://image.com/tata.jpg"
        />
      </div>
    </div>
  </section>
</div>
`;
